<a href="https://colab.research.google.com/github/fjnarvaezo1-dotcom/UNAD_TRABAJO_-TICA-Y-GOBIERNO-DE-DATOS/blob/main/Configuracion_Especializacion_Unad_%C3%89TICA_Y_GOBIERNO_DE_DATOS.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<html lang="es">
<head>
<meta charset="UTF-8">
<title>Información de contacto</title>
</head>
<body>
<div style="text-align:center;">
<img src=>
<p>
<strong>Francisco Javier Narváez Ortiz</strong><br>
Cédula: 1023915471
</p>

<p>
<a>href/www.unad.edu.co/
https://www.unad.edu.co/
</a>
</p>
</div>

</body>
</html>




## **Limpieza, Transformación y Estandarización de Datos de Leads e Información de Campañas**

### Dataframe numero 1 ajuste de informacion

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [7]:
ruta_archivo = "/content/drive/MyDrive/GESTIÓN DE PROYECTOS PARA LA ANALÍTICA - (203008074A_2204)/Leads_Detail.csv"

In [8]:
import pandas as pd
df = pd.read_csv(ruta_archivo, sep=',')
df = pd.DataFrame(df)
df.head(60000)

,lead_id,campaign_id,contact_date,responded,scheduled_demo,closed_deal,deal_value
0,LD001,CAM027,2024-02-01,True,False,True,9771.31
1,LD002,CAM055,2024-02-02,True,False,True,12446.13
2,LD003,CAM033,2024-02-03,True,False,False,3096.32
3,LD004,CAM068,2024-02-04,False,True,True,164.94
4,LD005,CAM066,2024-02-05,True,False,False,2053.28
...,...,...,...,...,...,...,...
75,LD076,CAM009,2024-04-16,True,True,True,8519.18
76,LD077,CAM074,2024-04-17,False,True,False,13732.32
77,LD078,CAM052,2024-04-18,True,True,False,509.19
78,LD079,CAM057,2024-04-19,True,True,True,10461.30


### Dataframe Numero 2

In [9]:
ruta = "/content/drive/MyDrive/GESTIÓN DE PROYECTOS PARA LA ANALÍTICA - (203008074A_2204)/campaign_performance_data.csv"

In [10]:
df_2 = pd.read_csv(ruta, sep=',')
df_2 = pd.DataFrame(df_2)
df_2.head(100)

,campaign_id,ad_platform,campaign_name,industry,start_date,end_date,impressions,clicks,cost_usd,conversions,conversion_value,leads,ctr,roas,region
0,CAM001,Meta Ads,Campaña ERP 1,Agroindustria,2024-01-01,2024-01-31,238576,1943,9950.95,100,20581.44,67,0.022,1.53,Argentina
1,CAM002,Google Ads,Campaña ERP 2,Cross-sector,2024-01-08,2024-02-07,239170,8780,5229.50,130,25876.02,53,0.033,4.07,Perú
2,CAM003,Meta Ads,Campaña ERP 3,Salud,2024-01-15,2024-02-14,23545,3076,3516.04,122,5341.45,182,0.016,3.79,Chile
3,CAM004,Meta Ads,Campaña ERP 4,Tecnología,2024-01-22,2024-02-21,137659,2059,8951.45,52,9500.56,169,0.058,1.09,Argentina
4,CAM005,Google Ads,Campaña ERP 5,HR Services,2024-01-29,2024-02-28,207271,5859,7729.47,93,20738.50,182,0.040,1.20,Perú
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
75,CAM076,LinkedIn,Campaña ERP 76,Agroindustria,2025-06-09,2025-07-09,236814,9107,4854.83,221,14607.58,32,0.041,2.64,Argentina
76,CAM077,LinkedIn,Campaña ERP 77,Finanzas,2025-06-16,2025-07-16,247714,7783,7196.50,281,17242.31,62,0.014,1.36,Colombia
77,CAM078,Meta Ads,Campaña ERP 78,Salud,2025-06-23,2025-07-23,152483,4742,1523.74,253,3818.30,174,0.054,0.75,Colombia
78,CAM079,Meta Ads,Campaña ERP 79,Cross-sector,2025-06-30,2025-07-30,143983,7779,9236.92,168,18964.58,211,0.056,3.13,Argentina


### Consolidacion de datos y analisis , se realizo una unificacion a continuacion se confirman los datos y se concatenan despues de ajustar el dataframe de los Datos.

In [11]:
derged = pd.merge(df, df_2, on='campaign_id', how='inner')
derged_2 = pd.DataFrame(derged)
derged_2.head(100)

,lead_id,campaign_id,contact_date,responded,scheduled_demo,closed_deal,deal_value,ad_platform,campaign_name,industry,...,end_date,impressions,clicks,cost_usd,conversions,conversion_value,leads,ctr,roas,region
0,LD001,CAM027,2024-02-01,True,False,True,9771.31,Google Ads,Campaña ERP 27,Agroindustria,...,2024-07-31,235281,2355,8638.03,104,4117.94,165,0.011,1.77,Chile
1,LD002,CAM055,2024-02-02,True,False,True,12446.13,LinkedIn,Campaña ERP 55,Agroindustria,...,2025-02-12,116081,6115,5833.74,272,30932.24,14,0.042,4.50,México
2,LD003,CAM033,2024-02-03,True,False,False,3096.32,LinkedIn,Campaña ERP 33,Agroindustria,...,2024-09-11,179695,1748,3305.86,14,4857.80,195,0.027,2.59,Chile
3,LD004,CAM068,2024-02-04,False,True,True,164.94,LinkedIn,Campaña ERP 68,Agroindustria,...,2025-05-14,101906,5995,7431.36,268,14776.87,186,0.016,4.66,Perú
4,LD005,CAM066,2024-02-05,True,False,False,2053.28,LinkedIn,Campaña ERP 66,Agroindustria,...,2025-04-30,61663,3536,2194.71,219,15771.36,226,0.019,1.02,Perú
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
75,LD076,CAM009,2024-04-16,True,True,True,8519.18,Meta Ads,Campaña ERP 9,Cross-sector,...,2024-03-27,71087,1902,6150.63,196,24114.72,212,0.028,1.97,México
76,LD077,CAM074,2024-04-17,False,True,False,13732.32,LinkedIn,Campaña ERP 74,Cross-sector,...,2025-06-25,134249,5700,2659.00,118,24155.34,142,0.012,1.75,Perú
77,LD078,CAM052,2024-04-18,True,True,False,509.19,Google Ads,Campaña ERP 52,Logística,...,2025-01-22,208935,5011,3252.26,37,8929.75,295,0.016,1.10,Argentina
78,LD079,CAM057,2024-04-19,True,True,True,10461.30,Meta Ads,Campaña ERP 57,Logística,...,2025-02-26,99045,4596,8450.82,172,31861.59,298,0.027,2.72,Argentina


### Consolidacion y limpieza de Datos los cuales se consolidaron para extraer las metricas de la informacion con lo cual se implemeto una IA, para realizar el analisis de los datos, en el siguiente dataframe se confirman los datos los cuales estan ya correlacionados y limpios para el proceso de analisis.

In [13]:
data = derged_2[["lead_id","campaign_id","closed_deal","responded","scheduled_demo","impressions","clicks","cost_usd","deal_value","region"]]
data = pd.DataFrame(data)
data.head(500)

,lead_id,campaign_id,closed_deal,responded,scheduled_demo,impressions,clicks,cost_usd,deal_value,region
0,LD001,CAM027,True,True,False,235281,2355,8638.03,9771.31,Chile
1,LD002,CAM055,True,True,False,116081,6115,5833.74,12446.13,México
2,LD003,CAM033,False,True,False,179695,1748,3305.86,3096.32,Chile
3,LD004,CAM068,True,False,True,101906,5995,7431.36,164.94,Perú
4,LD005,CAM066,False,True,False,61663,3536,2194.71,2053.28,Perú
...,...,...,...,...,...,...,...,...,...,...
75,LD076,CAM009,True,True,True,71087,1902,6150.63,8519.18,México
76,LD077,CAM074,False,False,True,134249,5700,2659.00,13732.32,Perú
77,LD078,CAM052,False,True,True,208935,5011,3252.26,509.19,Argentina
78,LD079,CAM057,True,True,True,99045,4596,8450.82,10461.30,Argentina


### Costo de las campañas y los leads por region, resumen ejecutivo despues de limpiar los datos.

In [17]:
information_coust = data.groupby(["region"])["cost_usd"].sum()
information_coust = pd.DataFrame(information_coust)
information_coust

,cost_usd
region,
Argentina,110977.91
Chile,50685.93
Colombia,67997.28
México,56892.61
Perú,101167.43


### Costo de los leads por region donde se concluye que el mas costoso es colombia a nivel de inversion

|region|cost\_usd|
|---|---|
|Argentina|110977\.91|
|Chile|50685\.93|
|Colombia|67997\.28|
|México|56892\.61|
|Perú|101167\.43|

### Ganancias Netas por region

In [15]:
information_negocio_cerrado_por_region = data.groupby(["region"])[["deal_value","closed_deal"]].sum()
information_negocio = pd.DataFrame(information_negocio_cerrado_por_region)
information_negocio



,deal_value,closed_deal
region,,
Argentina,138860.72,7
Chile,76775.70,4
Colombia,136048.82,7
México,93278.51,7
Perú,154912.40,12


### la IA solicito el valor "impressions" para darme un dato mas exacto en ese caso se le brindara la informacion

In [ ]:
impressions = data.groupby(["region"])["impressions"].sum()
impressions = pd.DataFrame(impressions)
impressions



,impressions
region,
Argentina,2872073
Chile,1528779
Colombia,3092958
México,1681284
Perú,3056714


## **Documentacion de entrenamiento de IA**

# Entrenamiento del Agente

## Rol
Experto en Ciencia de Datos, Analítica y Marketing Analytics.

# Objetivo
Identificar oportunidades para mejorar:
- ROAS
- ROI
- Rentabilidad
- Conversión comercial

## Métricas Obligatorias
- ROAS = Ingresos / Inversión
- ROI = (Ingresos - Costos) / Costos
- CPA = Inversión / Conversiones
- CTR = (Clicks / Impresiones) * 100

## Datos Utilizados
- Costos por región
- Valor de oportunidades
- Negocios cerrados
- Leads y campañas

## Dataframes de datos ajustados
**usa los datos que te comparto para realizar el analisis y brindarme los planes de mejora para el ROAS y las metricas de las campañas**

### Dataframe con las ganancias netas y los negocios cerrados por region

|region|deal\_value|closed\_deal|
|---|---|---|
|Argentina|138860\.72|7|
|Chile|76775\.7|4|
|Colombia|136048\.82|7|
|México|93278\.51|7|
|Perú|154912\.4|12|

### Dataframe de los costos de los leads en resumen por region

|region|cost\_usd|
|---|---|
|Argentina|110977\.91|
|Chile|50685\.93|
|Colombia|67997\.28|
|México|56892\.61|
|Perú|101167\.43|

### Dataframe con los datos limpios para procesamiento, este dataframe tiene la informacion de los leads y campañas resumidos con informacion limpia y ajustada

|index|lead\_id|campaign\_id|closed\_deal|responded|scheduled\_demo|impressions|clicks|cost\_usd|deal\_value|region|
|---|---|---|---|---|---|---|---|---|---|---|
|0|LD001|CAM027|true|true|false|235281|2355|8638\.03|9771\.31|Chile|
|1|LD002|CAM055|true|true|false|116081|6115|5833\.74|12446\.13|México|
|2|LD003|CAM033|false|true|false|179695|1748|3305\.86|3096\.32|Chile|
|3|LD004|CAM068|true|false|true|101906|5995|7431\.36|164\.94|Perú|
|4|LD005|CAM066|false|true|false|61663|3536|2194\.71|2053\.28|Perú|
|5|LD006|CAM010|false|false|false|236911|8255|9822\.98|13500\.28|Argentina|
|6|LD007|CAM005|false|true|false|207271|5859|7729\.47|13108\.35|Perú|
|7|LD008|CAM074|true|false|true|134249|5700|2659\.0|8961\.2|Perú|
|8|LD009|CAM038|true|false|true|60859|2657|2748\.47|9007\.75|Argentina|
|9|LD010|CAM013|true|true|false|195456|5634|2718\.2|9975\.55|México|
|10|LD011|CAM031|false|false|true|141926|6297|3668\.59|2630\.57|Colombia|
|11|LD012|CAM047|true|true|false|119751|8452|3953\.37|13716\.18|Perú|
|12|LD013|CAM052|false|false|true|208935|5011|3252\.26|6281\.56|Argentina|
|13|LD014|CAM056|true|true|true|206805|1318|3939\.86|5747\.08|Perú|
|14|LD015|CAM015|false|true|true|187648|362|5367\.52|7783\.77|México|
|15|LD016|CAM029|true|false|true|122859|4599|7380\.2|704\.49|Argentina|
|16|LD017|CAM008|true|true|false|103557|4836|5974\.88|2494\.25|México|
|17|LD018|CAM005|true|true|false|207271|5859|7729\.47|11070\.5|Perú|
|18|LD019|CAM029|false|false|false|122859|4599|7380\.2|1241\.98|Argentina|
|19|LD020|CAM047|false|true|true|119751|8452|3953\.37|9047\.28|Perú|
|20|LD021|CAM068|true|true|false|101906|5995|7431\.36|3680\.24|Perú|
|21|LD022|CAM076|true|false|false|236814|9107|4854\.83|5839\.43|Argentina|
|22|LD023|CAM045|false|false|false|193734|2969|4946\.27|4330\.41|Argentina|
|23|LD024|CAM002|false|true|true|239170|8780|5229\.5|5335\.09|Perú|
|24|LD025|CAM027|false|true|true|235281|2355|8638\.03|10785\.69|Chile|
|25|LD026|CAM036|true|false|true|187789|4299|1046\.66|4456\.83|Colombia|
|26|LD027|CAM036|false|false|false|187789|4299|1046\.66|8496\.07|Colombia|
|27|LD028|CAM026|true|true|false|36736|9974|3902\.96|7140\.76|Chile|
|28|LD029|CAM043|false|false|false|204806|5073|1248\.55|9955\.07|Perú|
|29|LD030|CAM027|true|true|true|235281|2355|8638\.03|14052\.45|Chile|
|30|LD031|CAM069|false|true|true|59811|3454|1369\.61|10988\.58|México|
|31|LD032|CAM020|false|true|false|75318|5523|1633\.75|3224\.11|Colombia|
|32|LD033|CAM011|false|true|true|78840|8220|1678\.12|467\.75|México|
|33|LD034|CAM074|false|true|true|134249|5700|2659\.0|3933\.96|Perú|
|34|LD035|CAM038|false|false|true|60859|2657|2748\.47|8926\.17|Argentina|
|35|LD036|CAM006|true|false|true|175838|6794|9577\.65|771\.39|Perú|
|36|LD037|CAM072|false|false|false|122547|7122|7696\.38|7445\.49|Colombia|
|37|LD038|CAM023|false|false|true|77172|7258|4227\.82|8952\.64|Perú|
|38|LD039|CAM047|false|true|true|119751|8452|3953\.37|5013\.66|Perú|
|39|LD040|CAM046|true|true|true|108506|1056|7048\.24|11563\.68|Argentina|
|40|LD041|CAM012|false|true|true|128451|6716|3751\.27|1598\.97|Chile|
|41|LD042|CAM013|true|true|false|195456|5634|2718\.2|1127\.07|México|
|42|LD043|CAM062|false|false|false|166256|3546|2749\.42|10922\.83|Colombia|
|43|LD044|CAM060|false|true|true|90623|9246|5115\.39|7432\.37|Colombia|
|44|LD045|CAM043|true|true|true|204806|5073|1248\.55|10326\.04|Perú|
|45|LD046|CAM076|false|false|true|236814|9107|4854\.83|6522\.41|Argentina|
|46|LD047|CAM068|true|false|true|101906|5995|7431\.36|3696\.03|Perú|
|47|LD048|CAM005|true|false|true|207271|5859|7729\.47|12286\.53|Perú|
|48|LD049|CAM037|false|false|true|228164|8545|6651\.05|11991\.24|Chile|
|49|LD050|CAM072|true|true|false|122547|7122|7696\.38|10420\.45|Colombia|
|50|LD051|CAM031|false|true|true|141926|6297|3668\.59|4082\.18|Colombia|
|51|LD052|CAM009|false|false|false|71087|1902|6150\.63|8853\.46|México|
|52|LD053|CAM051|true|true|false|178229|9592|8743\.64|5414\.61|Colombia|
|53|LD054|CAM029|true|true|true|122859|4599|7380\.2|1373\.73|Argentina|
|54|LD055|CAM078|true|true|false|152483|4742|1523\.74|13759\.7|Colombia|
|55|LD056|CAM040|true|false|false|245167|198|4571\.05|2052\.28|Colombia|
|56|LD057|CAM041|false|true|false|167381|2300|1456\.92|14253\.56|Argentina|
|57|LD058|CAM011|false|false|false|78840|8220|1678\.12|6690\.09|México|
|58|LD059|CAM023|true|true|false|77172|7258|4227\.82|2776\.99|Perú|
|59|LD060|CAM001|false|true|false|238576|1943|9950\.95|8128\.51|Argentina|
|60|LD061|CAM046|false|true|true|108506|1056|7048\.24|13094\.19|Argentina|
|61|LD062|CAM021|false|false|false|230546|4648|2880\.27|10983\.37|Colombia|
|62|LD063|CAM036|false|true|false|187789|4299|1046\.66|12098\.42|Colombia|
|63|LD064|CAM054|true|false|false|193155|9819|3729\.39|9881\.75|Colombia|
|64|LD065|CAM057|true|false|true|99045|4596|8450\.82|10384\.15|Argentina|
|65|LD066|CAM001|false|false|false|238576|1943|9950\.95|12737\.93|Argentina|
|66|LD067|CAM063|true|true|true|103384|1145|4702\.19|3745\.02|México|
|67|LD068|CAM054|true|true|false|193155|9819|3729\.39|7341\.37|Colombia|
|68|LD069|CAM055|true|true|true|116081|6115|5833\.74|3318\.14|México|
|69|LD070|CAM040|false|true|true|245167|198|4571\.05|14815\.02|Colombia|
|70|LD071|CAM015|false|false|false|187648|362|5367\.52|14160\.89|México|
|71|LD072|CAM021|false|false|false|230546|4648|2880\.27|591\.4|Colombia|
|72|LD073|CAM047|true|false|true|119751|8452|3953\.37|10583\.63|Perú|
|73|LD074|CAM073|false|true|false|213154|9251|3257\.74|13878\.72|Chile|
|74|LD075|CAM053|false|false|true|116308|4087|1349\.51|2708\.63|México|
|75|LD076|CAM009|true|true|true|71087|1902|6150\.63|8519\.18|México|
|76|LD077|CAM074|false|false|true|134249|5700|2659\.0|13732\.32|Perú|
|77|LD078|CAM052|false|true|true|208935|5011|3252\.26|509\.19|Argentina|
|78|LD079|CAM057|true|true|true|99045|4596|8450\.82|10461\.3|Argentina|
|79|LD080|CAM026|true|false|false|36736|9974|3902\.96|4460\.24|Chile|

## Metodología
1. Calcular ROAS actual.
2. Analizar ROAS por región y campaña.
3. Detectar campañas con ROAS < 1.
4. Identificar CPA altos.
5. Redistribuir presupuesto hacia campañas rentables.
6. Optimizar el funnel:
Impresión → Click → Respuesta → Demo → Cierre.

## Recomendaciones Esperadas
- Pausar campañas deficitarias.
- Escalar campañas rentables.
- Mejorar conversión Demo → Cierre.
- Aumentar ticket promedio.
- Mejorar asignación del presupuesto.

## Objetivo Final
Aumentar el ROAS y el retorno de ganancia sin incrementar significativamente la inversión.